tests/integration/customerPartners.test.js

```diff
@@ -0,0 +1,215 @@
+    });
+
+    it('should return 403 if partner has wrong company', async () => {
+      const response = await app.inject({
+        method: 'POST',
+        url: '/customerpartners',
+        headers: { Cookie: `alenvi_token=${authToken}` },
+        payload: { customer: customersList[0]._id, partner: partnersList[1]._id },
+      });
+
+      expect(response.statusCode).toBe(403);
+    });
+
+    it('should return 403 if user has wrong company', async () => {
+      authToken = await getTokenByCredentials(auxiliaryFromOtherCompany.local);
+      const response = await app.inject({
+        method: 'POST',
+        url: '/customerpartners',
+        headers: { Cookie: `alenvi_token=${authToken}` },
+        payload: { customer: customersList[0]._id, partner: partnersList[0]._id },
+      });
+
+      expect(response.statusCode).toBe(403);
+    });
+
+    it('should return 409 if customer partner already exists', async () => {
+      authToken = await getTokenByCredentials(auxiliaryFromOtherCompany.local);
+      const response = await app.inject({
+        method: 'POST',
+        url: '/customerpartners',
+        headers: { Cookie: `alenvi_token=${authToken}` },
+        payload: { customer: customersList[1]._id, partner: partnersList[1]._id },
+      });
+
+      expect(response.statusCode).toBe(409);
+    });
+  });
+
+  describe('Other roles', () => {
+    const roles = [
+      { name: 'vendor_admin', expectedCode: 403 },
+      { name: 'helper', expectedCode: 403 },
+    ];
+
+    roles.forEach((role) => {
+      it(`should return ${role.expectedCode} as user is ${role.name}`, async () => {
+        authToken = await getToken(role.name);
+        const response = await app.inject({
+          method: 'POST',
+          url: '/customerpartners',
+          headers: { Cookie: `alenvi_token=${authToken}` },
+          payload: { customer: customersList[0]._id, partner: partnersList[0]._id },
+        });
+
+        expect(response.statusCode).toBe(role.expectedCode);
+      });
+    });
+  });
+});
+
+describe('CUSTOMER PARTNERS ROUTES - GET /customerpartners', () => {
+  let authToken = null;
+  beforeEach(populateDB);
+
+  describe('AUXILIARY', () => {
+    beforeEach(async () => {
+      authToken = await getToken('auxiliary');
+    });
+
+    it('should get all partners of a user', async () => {
+      const response = await app.inject({
+        method: 'GET',
+        url: `/customerpartners?customer=${customersList[0]._id}`,
+        headers: { Cookie: `alenvi_token=${authToken}` },
+      });
+
+      expect(response.statusCode).toBe(200);
+    });
+
+    it('should return 400 if query customer has invalid type', async () => {
+      const response = await app.inject({
+        method: 'GET',
+        url: '/customerpartners?customer=test',
+        headers: { Cookie: `alenvi_token=${authToken}` },
+      });
+
+      expect(response.statusCode).toBe(400);
+    });
+
+    it('should return 404 if invalid customer', async () => {
```

Comment:
customer does not exists pour etre plus precis

Reply:
done